packages/core/src/components/graphs/confidence-interval.ts

```diff
@@ -0,0 +1,208 @@
+// Internal Imports
+import { Component } from "../component";
+import * as Configuration from "../../configuration";
+import {
+	CartesianOrientations,
+	Events,
+	ColorClassNameTypes
+} from "../../interfaces";
+import { DOMUtils } from "../../services";
+import { Tools } from "../../tools";
+import settings from "carbon-components/es/globals/js/settings";
+
+// D3 Imports
+import { area } from "d3-shape";
+import { select } from "d3-selection";
+
+export class ConfidenceInterval extends Component {
+	type = "area";
+
+	init() {
+		const eventsFragment = this.services.events;
+
+		// Highlight correct area on legend item hovers
+		eventsFragment.addEventListener(
+			Events.Legend.ITEM_HOVER,
+			this.handleLegendOnHover
+		);
+
+		// Un-highlight area on legend item mouseouts
+		eventsFragment.addEventListener(
+			Events.Legend.ITEM_MOUSEOUT,
+			this.handleLegendMouseOut
+		);
+	}
+
+	render(animate = true) {
+		const svg = this.getContainerSVG({ withinChartClip: true });
+		let domain = [0, 0];
+
+		const { cartesianScales } = this.services;
+
+		const orientation = cartesianScales.getOrientation();
+		const areaGenerator = area().curve(this.services.curves.getD3Curve());
+
+		// Update the bound data on area groups
+		const groupedData = this.model.getGroupedData();
+
+		const confidence = groupedData && groupedData.length === 1;
+
+		if (!confidence) {
+			console.warn(
+				"Confidence Intervals can only be shown when having 1 single dataset"
+			); // eslint-disable-line no-console
+		}
+
+		if (orientation === CartesianOrientations.VERTICAL) {
+			areaGenerator
+				.x((d, i) => cartesianScales.getDomainValue(d, i))
+				.y0((d, i) =>
```

Comment:
can we move these into a variable? e.g.

```js
const yAccessor = confidence ? cartesianScales.getConfidenceScaledValues : cartesianScales.getRangeValue;
```

and then calling `yAccessor(d, i)` etc...

Reply:
Something similar to what you've been suggested should be in!